cloud/google/machineactuator.go

```diff
@@ -727,6 +687,74 @@ func getSubnet(netRange clusterv1.NetworkRanges) string {
 	return netRange.CIDRBlocks[0]
 }
 
+func getOrNewComputeService(params MachineActuatorParams) (GCEClientComputeService, error) {
+	if params.ComputeService != nil {
+		return params.ComputeService, nil
+	}
+	// The default GCP client expects the environment variable
+	// GOOGLE_APPLICATION_CREDENTIALS to point to a file with service credentials.
+	client, err := google.DefaultClient(context.TODO(), compute.ComputeScope)
+	if err != nil {
+		return nil, err
+	}
+	computeService, err := clients.NewComputeService(client)
+	if err != nil {
+		return nil, err
+	}
+	return computeService, nil
+}
+
+func (gce *GCEClient) getMetadata(cluster *clusterv1.Cluster, machine *clusterv1.Machine, config *gceconfigv1.GCEProviderConfig, configParams *machinesetup.ConfigParams, ca *cert.CertificateAuthority) (*compute.Metadata, error) {
```

Comment:
The reason why it is passed in is because it is still being used in Create(...) to call "GetImage(configParams)". 

Do you think it would be better to break out assembly of the object to a function and have both Create(...) and getMetadata(...) use that function and have separate copies or to keep it as it is and pass it in? I see merit to both options.

Reply:
Ah I missed the call to `GetImage(configParams)`. I think it's fine to just keep it as is and pass the configParams in then.